lib/matplotlib/figure.py

```diff
@@ -1173,12 +1173,14 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
         if sharex in ["col", "all"]:
             # turn off all but the bottom row
             for ax in axarr[:-1, :].flat:
-                ax.xaxis.set_tick_params(labelbottom=False)
+                ax.xaxis.set_tick_params(which='both',
```

Comment:
When I wrote this I went back and forth on if removing _all_ of the labels vs just the bottom and left was overkill.

'labelbottom' and 'labelleft' are certainly more readable!